integration_test/test_guest_login.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:dairy_app/app/view/app.dart';
+import 'package:dairy_app/core/pages/home_page.dart';
+import 'package:dairy_app/features/auth/presentation/pages/auth_page.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+Future<void> main() async{
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
```

Comment:
@PratyushChauhan can you try calling `di.init()` as well here
```
import 'core/dependency_injection/injection_container.dart' as di;

...
  await di.init();

```

Reply:
tried this, got an async gap error.
```
PS C:\dev\projects\OSS\diaryvault> flutter test integration_test/test_guest_login.dart                                                                                                                                                                                 
00:00 +0: loading C:/dev/projects/OSS/diaryvault/integration_test/test_guest_login.dart                                                                                                                                                                                             R02:00 +0: loading C:/dev/projects/OSS/diaryvault/integration_test/test_guest_login.dart                                                                                                                                                                                       118.6s
√  Built build\app\outputs\flutter-apk\app-debug.apk.
02:04 +0: loading C:/dev/projects/OSS/diaryvault/integration_test/test_guest_login.dart                                                                                                                                                                                             I02:12 +0: loading C:/dev/projects/OSS/diaryvault/integration_test/test_guest_login.dart                                                                                                                                                                                         7.9s
02:17 +0: loading C:/dev/projects/OSS/diaryvault/integration_test/test_guest_login.dart                                                                                                                                                                                             
🐛  AuthSessionBloc: AuthSessionBloc recreated
💡  KeyValueDataSource: searching for key current_theme
02:17 +0 -1: loading C:/dev/projects/OSS/diaryvault/integration_test/test_guest_login.dart [E]                                                                                                                                                                                      
  Failed to load "C:/dev/projects/OSS/diaryvault/integration_test/test_guest_login.dart": Instance of 'NotInitializedError'
  package:flutter_dotenv/src/dotenv.dart 41:7                                                                 DotEnv.env
  package:dairy_app/features/sync/data/datasources/dropbox_sync_client.dart 23:39                             new DropboxSyncClient
  package:dairy_app/core/dependency_injection/injection_container.dart 135:7                                  init
  ===== asynchronous gap ===========================
  c:/dev/projects/OSS/diaryvault/integration_test/test_guest_login.dart 10:3                                  main
  ===== asynchronous gap ===========================
  c:/Users/ciphe/AppData/Local/Temp/flutter_tools.5a76b99f/flutter_test_listener.48a70eb7/listener.dart 22:3  _testMain


To run this test again: C:\dev\tooling\flutter\bin\cache\dart-sdk\bin\dart.exe test C:/dev/projects/OSS/diaryvault/integration_test/test_guest_login.dart -p vm --plain-name "loading C:/dev/projects/OSS/diaryvault/integration_test/test_guest_login.dart"
02:17 +0 -1: Some tests failed.

```

Reply:
updated function
```
Future<void> main() async{
  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
  await di.init();


  testWidgets('Guest login test, click on sign up button to move reveal guest login button'
      'click the guest login button to enter HomePage', (WidgetTester tester) async {
    await tester.pumpWidget(App());
    await tester.pumpAndSettle();
    await tester.tap(find.text('Sign up'));
    await tester.pumpAndSettle();
    await tester.tap(find.text('Continue as guest'));
    await tester.pumpAndSettle();
    expect(find.byType(HomePage), findsOneWidget);
    expect(find.byType(AuthPage), findsNothing);
  });
}
```

Reply:
Hey sorry for late reply, try calling `di.init` in `setUpAll` function. ref: https://stackoverflow.com/questions/62548585/how-to-use-get-it-flutter-dependency-injection-in-the-widget-test